osidb/tests/test_flaw.py

```diff
@@ -1886,6 +1889,14 @@ def test_validate_flaw_without_affect(self):
             flaw2.save()
         assert "Flaw does not contain any affects." in str(e)
 
+        # Flaws in the 'new' state can be edited without any affects, but
```

Comment:
In the old code above there is twice created a flaw by the factory without providing any `workflow_state` value which seems to result in a weird situation when there is an empty value as it has no default but the empty value is not among the choices. I am surprised that it is even possible and therefore I suspect that the `ValidationError` might actually be raised because of this and not because of the missing affect (or there are two errors at the same time). I would provide a value different from `NEW` to prevent the weirdness.

Reply:
It is indeed created with an empty state, and any state other than `NEW` raises the error. I will make it have another state to make it more clear.